src/daft-connect/src/op/execute/root.rs

```diff
@@ -0,0 +1,107 @@
+use std::{collections::HashMap, future::ready};
+
+use common_daft_config::DaftExecutionConfig;
+use futures::stream;
+use spark_connect::{ExecutePlanResponse, Relation};
+use tokio_util::sync::CancellationToken;
+use tonic::{codegen::tokio_stream::wrappers::UnboundedReceiverStream, Status};
+
+use crate::{
+    op::execute::{ExecuteStream, PlanIds},
+    session::Session,
+    translation,
+};
+
+impl Session {
+    pub async fn handle_root_command(
+        &self,
+        command: Relation,
+        operation_id: String,
+    ) -> Result<ExecuteStream, Status> {
+        use futures::{StreamExt, TryStreamExt};
+
+        let context = PlanIds {
+            session: self.client_side_session_id().to_string(),
+            server_side_session: self.server_side_session_id().to_string(),
+            operation: operation_id,
+        };
+
+        let finished = context.finished();
+
+        let (tx, rx) = tokio::sync::mpsc::unbounded_channel::<eyre::Result<ExecutePlanResponse>>();
+
+        std::thread::spawn(move || {
+            let plan = match translation::to_logical_plan(command) {
```

Comment:
this body can instead be a function that returns a result. and that can be called and fed into the channel rather than every `?` equivalent having to be handled independently

Reply:
c8688f14c15dbe316e39952a46cbc3257b02e388